lib/administrate/generator_helpers.rb

```diff
@@ -1,5 +1,17 @@
+require "find"
+
 module Administrate
   module GeneratorHelpers
+    DEFAULT_INDENT = 2
+    ONE_INDENT = 1
+    TWO_INDENT = 2
```

Comment:
It is being used as the standard unit of indent, perhaps naming it as `STARDARD_INDENT` would be better?

Reply:
Oh, I see. I had misunderstood it. The name is ok then (although `STANDARD_INDENT`) also works.

I was a bit confused as to what `ONE_INDENT` and `TWO_INDENT` meant. I get them now. I think I would have implemented this differently, in a way that avoids the index wrangling and removes the need for these two constants. The main idea would be not to send the whole list on to the next recursive method call, but only the bits of the list that remain to be rendered. This is a common pattern in functional programming. Let me have a try...

Reply:
Actually, before I try to reimplement this, I found a new bug. If there is more than one namespaced resource, each will get its own `namespace` entry, like this:
```ruby
    namespace :blog do
      resources :posts
    end
    namespace :blog do
      resources :users
    end
```

To solve this, I suggest that `generate_resource_routes` takes the whole list of resources, instead of one at a time. Then it'll have to change to handle this list.

Reply:
That makes sense, we'll look into this tomorrow. Just for my clarity would this break any functionality or is it just badly formatted?

Reply:
It should technically work; it's just unexpected (and arguably bad) formatting.

Reply:
should be fixed, any advice for a good way to test this - as the current system in the `routes_generator_spec` uses the example app models (which don't have extended namespace structure.